mlx_vlm/models/fastvlm/fastvlm.py

```diff
@@ -0,0 +1,188 @@
+
+        # Encode images
+        _, image_features, _ = self.vision_tower(pixel_values.transpose(0, 2, 3, 1))
+        B, H, W, C = image_features.shape
+        image_features = image_features.reshape(B, H * W, C)
+        image_features = self.mm_projector(image_features)
+
+        final_inputs_embeds = self.prepare_inputs_for_multimodal(
+            image_features, input_ids
+        )
+        return final_inputs_embeds
+
+    # Adapted from https://github.com/apple/ml-fastvlm/blob/592b4add3c1c8a518e77d95dc6248e76c1dd591f/llava/model/llava_arch.py#L146
+    def prepare_inputs_for_multimodal(self, image_features, input_ids):
+        # TODO: padding, attention_mask
+
+        new_input_embeds = []
+        cur_image_idx = 0
+        for batch_idx, cur_input_ids in enumerate(input_ids):
+            num_images = (cur_input_ids == self.config.image_token_index).sum()
+            if num_images == 0:
+                cur_image_features = image_features[cur_image_idx]
+                cur_input_embeds_1 = self.language_model.model.embed_tokens(
+                    cur_input_ids
+                )
+                cur_input_embeds = mx.concatenate(
+                    [cur_input_embeds_1, cur_image_features[0:0]], dim=0
+                )
+                new_input_embeds.append(cur_input_embeds)
+                cur_image_idx += 1
+                continue
+
+            # Go to numpy for np.where() with a single arg (returns the matching positions)
+            image_token_indices = (
+                [-1]
+                + np.where(np.array(cur_input_ids == self.config.image_token_index))[
+                    0
+                ].tolist()
+                + [cur_input_ids.shape[0]]
+            )
+            cur_input_ids_noim = []
+            for i in range(len(image_token_indices) - 1):
+                cur_input_ids_noim.append(
+                    cur_input_ids[
+                        image_token_indices[i] + 1 : image_token_indices[i + 1]
+                    ]
+                )
+            split_sizes = image_token_indices[
+                1:
+            ]  # [x.shape[0] for x in cur_input_ids_noim]
+            cur_input_embeds = self.language_model.model.embed_tokens(
+                mx.concatenate(cur_input_ids_noim)
+            )
+            cur_input_embeds_no_im = mx.split(cur_input_embeds, split_sizes)
+
+            cur_new_input_embeds = []
+            for i in range(num_images.item() + 1):
+                cur_new_input_embeds.append(cur_input_embeds_no_im[i])
+                if i < num_images:
+                    cur_image_features = image_features[cur_image_idx]
+                    cur_image_idx += 1
+                    cur_new_input_embeds.append(cur_image_features)
+            cur_new_input_embeds = mx.concatenate(cur_new_input_embeds)
+
+            new_input_embeds.append(cur_new_input_embeds)
+
+        # Truncate sequences to max length as image embeddings can make the sequence longer
```

Comment:
```suggestion
```

Reply:
I actually find this one helpful for readers of the code, but removing